### PR TITLE
feat: make alchemy-sdk an optional dependency on aa-alchemy

### PR DESCRIPTION
**Pull request by avasisht23**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, see guidleines below)
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:fix`)
- [x] Is the base branch you're merging into `development` and not `main`?

---

### Detailed summary

- Updated `alchemy-sdk` dependency from version 2.8.3 to 3.0.0.
- Changed `alchemy-sdk` from a peer dependency to an optional dependency.
- Updated documentation to reflect the changes in the dependency.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by avasisht23** [comment]
Current dependencies on/for this PR:
* development
  * **PR #250** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/250?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #254** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/254?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #255** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/255?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #256** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/256?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #265** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/265?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/265?utm_source=stack-comment).

**Review by moldy530 — COMMENTED** [review]
There are a lot of other changes in here than just updating the dep, is that intentional?

File: site/overview/getting-started.md [hunk 1]
```diff
@@ -20,48 +20,172 @@ head:
...
+  "compilerOptions": {
+    "target": "ES2022",
+    "module": "ES2022",
+    "moduleResolution": "node",
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "skipLibCheck": true
+  }
+}
```
Comment (denniswon) [inline site/overview/getting-started.md:84]: where do these config requirements come from?

File: site/overview/getting-started.md [hunk 2]
```diff
@@ -20,48 +20,172 @@ head:
...
+```bash [nvm]
+nvm install 18.10.0
+nvm use 18.10.0
+```
+
+```bash [asdf]
+asdf install nodejs 18.10.0
+asdf global nodejs 18.10.0
+asdf reshim nodejs
+```
```
Comment (denniswon) [inline site/overview/getting-started.md:102]: ditto: what should devs whose projects use some conflicting configuration?

File: site/overview/getting-started.md [hunk 3]
```diff
@@ -20,48 +20,172 @@ head:
...
+    "skipLibCheck": true
+  }
+}
+```
 
-::: tip Note
-Remember to:
+:::
+
+You'll also want to make sure your `Node` version is _18.10.0_ using your version manager:
```
Comment (moldy530) [inline site/overview/getting-started.md:89]: isn't this being changed to 18.16?

File: site/overview/getting-started.md (hunk 1 above)
Comment (moldy530) [reply site/overview/getting-started.md:84]: wait why are we suggesting tsconfigs for devs? that shouldn't be necessary

File: site/overview/getting-started.md [hunk 4]
```diff
@@ -20,48 +20,172 @@ head:
...
 :::
 
-## A Simple Light Account Example
+Make sure your new `package.json` file and `tsconfig.json` files look like the following. Note that we added `"type": module` in `package.json` for this example:
 
-Using the SDK, we'll deploy a Light Account and send a User Operation from it. The Light Account will be owned by an Externally Owned Account (EOA). Here's a demonstration:
+::: code-group
 
-<<< @/snippets/light-account.ts
+```json [package.json]
```
Comment (moldy530) [inline site/overview/getting-started.md:53]: why are we recommending tsconfig and package.json for peeps? if our library doesn't work with a range of tsconfigs then we broke something I think. 

the package.json is confusing here too... we already told them to `yarn init` and `yarn add`? so this is a given

File: site/overview/getting-started.md (hunk 2 above)
Comment (moldy530) [reply site/overview/getting-started.md:102]: yea I think this whole section should be out of scope for this doc... it's not on us to explain tsconfig, asdf / nvm, or even package.json configs....

if we require a specific tsconfig, then we've done something wrong and we should fix it

File: site/overview/getting-started.md (hunk 4 above)
Comment (avasisht23) [reply site/overview/getting-started.md:53]: You need to add "type: module" still

File: site/overview/getting-started.md (hunk 1 above)
Comment (avasisht23) [reply site/overview/getting-started.md:84]: Agree sorry will remove

File: site/overview/getting-started.md (hunk 3 above)
Comment (avasisht23) [reply site/overview/getting-started.md:89]: After aa-signers will change to 18.16, not needed right now

File: site/overview/getting-started.md (hunk 2 above)
Comment (avasisht23) [reply site/overview/getting-started.md:102]: @avarobinson let me know if you feel otherwise

Comment (avasisht23) [reply site/overview/getting-started.md:102]: But I am removing tsconfig

File: site/overview/getting-started.md (hunk 4 above)
Comment (avasisht23) [reply site/overview/getting-started.md:51]: I did change the npm flag to be `npm init es6 -y` so it's there by default. I just think we should be super clear on each step so devs get unblocked, accordingly to feedback from Ava.

**Review by moldy530 — APPROVED** [review]

**Pull request merged**